File: code/library/src/main/java/org/fbme/lib/iec61499/declarations/ParameterDeclaration.java
package org.fbme.lib.iec61499.declarations;

import org.fbme.lib.common.ContainedElement;
import org.fbme.lib.common.Declaration;
import org.fbme.lib.st.expressions.Literal;
import org.fbme.lib.st.expressions.VariableDeclaration;
import org.fbme.lib.st.types.DataType;
import org.jetbrains.annotations.Nullable;

public interface ParameterDeclaration extends VariableDeclaration, ContainedElement {

    @Nullable Declaration getContainer();

    @Nullable DataType getType();

    void setType(@Nullable DataType type);

    @Nullable Literal<?> getInitialValue();

    void setInitialValue(@Nullable Literal<?> literal);
}

File: code/library/src/main/kotlin/org/fbme/lib/iec61499/declarations/ParameterDeclaration.kt
package org.fbme.lib.iec61499.declarations

import org.fbme.lib.common.ContainedElement
import org.fbme.lib.common.Declaration
import org.fbme.lib.st.expressions.Literal
import org.fbme.lib.st.expressions.VariableDeclaration
import org.fbme.lib.st.types.DataType

interface ParameterDeclaration : VariableDeclaration, ContainedElement {
    override val container: Declaration?
    var type: DataType?
    var initialValue: Literal<*>?
}
